test/test_xarray_io.py

```diff
@@ -0,0 +1,385 @@
+    data = xr.DataArray(da.arange(0, max_size, 1), dims="x")
+    data = data.chunk(chunks)
+    file_path = tmp_path / "test_xarray_to_text_jinja.disu"
+
+    profile_function(
+        create_and_write_jinja,
+        (file_path, data),
+        time_file,
+        print_args={"max_size": max_size, "chunks": chunks},
+    )
+
+    with open(file_path, "r") as f:
+        output = f.readlines()
+        assert (
+            len(output) == 2 + max_size / chunks
+        )  # begin + end + lines of data
+
+
+def create_and_write_pandas(file_path, data: xr.DataArray):
+    pandas_data = data.to_pandas()
+    with open(file_path, "w") as f:
+        f.write("BEGIN GRIDDATA\n")
+        pandas_data.to_csv(
+            f,
+            header=False,
+            index=False,
+            lineterminator=" ",
+            float_format="%.4f",
+        )
+        f.write("\nEND GRIDDATA\n")
+
+
+@pytest.mark.parametrize("max_size,chunks", test_combinations)
+@pytest.mark.timing
+def test_xarray_to_text_pandas(tmp_path, max_size, chunks, time_file):
+    data = xr.DataArray(da.arange(0, max_size, 1), dims="x")
+    data = data.chunk(chunks)
+    file_path = tmp_path / "test_xarray_to_text_extras.disu"
+
+    profile_function(
+        create_and_write_pandas,
+        (file_path, data),
+        time_file,
+        print_args={"max_size": max_size, "chunks": chunks},
+    )
+
+    with open(file_path, "r") as f:
+        output = f.readlines()
+        assert len(output) == 3  # begin + end + 1 line of data
+
+
+def create_and_write_np_savetxt(file_path, data: xr.DataArray):
+    with open(file_path, "w") as f:
+        f.write("BEGIN GRIDDATA\n")
+        for block in data.data.to_delayed():
+            block_data = block.compute()
+            np.savetxt(f, block_data, newline=" ", fmt="%.4f")
+        f.write("\nEND GRIDDATA\n")
+
+
+@pytest.mark.parametrize("max_size,chunks", test_combinations)
+@pytest.mark.timing
+def test_xarray_to_text_np_savetxt(tmp_path, max_size, chunks, time_file):
+    data = xr.DataArray(da.arange(0, max_size, 1), dims="x")
+    data = data.chunk(chunks)
+    file_path = tmp_path / "test_xarray_to_text_raw.disu"
+
+    profile_function(
+        create_and_write_np_savetxt,
+        (file_path, data),
+        time_file,
+        print_args={"max_size": max_size, "chunks": chunks},
+    )
+
+    with open(file_path, "r") as f:
+        output = f.readlines()
+        assert len(output) == 3
+
+
+def create_and_write_extras(file_path, data: xr.DataArray):
+    with open(file_path, "w") as f:
+        f.write("BEGIN GRIDDATA\n")
+    promise = xarray_extras.csv.to_csv(
+        data,
+        file_path,
+        header=False,
+        index=False,
+        float_format="%.4f",
+        lineterminator=" ",
+        compression=None,
+        mode="a",
+    )
+    promise.compute()
+    # we have to open the file again,
+    # because xarray_extras only allows paths and no file handlers.
+    with open(file_path, "a") as f:
```

Comment:
one option would be to just use pandas `to_csv` for internal arrays, and accept the performance hit, and advise that for fast IO people should use external arrays. pandas accepts a file handle not only a path. is dask lazy io the reason extras does not? I wonder if extras could support file handle for the numpy-backed array case, then we can still get a speedup.